app/code/Magento/CatalogGraphQl/etc/graphql/di.xml

```diff
@@ -6,6 +6,7 @@
  */
 -->
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
+    <preference for="Magento\Catalog\Model\Product\Option\Type\Date" type="Magento\CatalogGraphQl\Model\Product\Option\DateType" />
```

Comment:
This looks risky because we override the default behavior of catalog class. Why is this needed?

Reply:
@paliarush It's needed for API purpose only (that's why it's defined in scope of graphql area) to bypass requirement for date values which Magento has on the frontend (date value should be an array). So by this, we made it possible to receive and validate data value in string format. 

Does it make sense?